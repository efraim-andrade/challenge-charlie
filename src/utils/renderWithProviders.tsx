import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import { theme } from '~/styles/theme/index';

export const renderWithProviders = (children: JSX.Element) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
};

export default renderWithProviders;
