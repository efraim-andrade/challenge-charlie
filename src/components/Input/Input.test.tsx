import React from "react";
import { fireEvent, screen } from '@testing-library/react';

import CompassIcon from '~/assets/compass.svg';
import renderWithProviders from '~/utils/renderWithProviders';

import { Input } from '.';

describe('components - <Input />', () => {
  it('should render correctly', () => {
    const { container } = renderWithProviders(<Input />);

    expect(container).toMatchSnapshot();
  });

  it('should render with an icon', () => {
    renderWithProviders(<Input icon={{ svg: CompassIcon, alt: 'Ícone de compasso' }} />);

    expect(screen.getByLabelText('Ícone de compasso')).toBeInTheDocument();
  });

  it('should handle with change event', () => {
    const onChange = jest.fn();

    renderWithProviders(<Input onChange={onChange} />);

    const input = screen.getByRole('textbox');

    fireEvent.change(input, { target: { value: 'John Doe' } });

    expect(onChange).toHaveBeenCalled();
    expect((input as HTMLInputElement).value).toBe('John Doe');
  });

  it('should handle focus on input', () => {
    const inputRef = React.createRef<HTMLInputElement>();

    renderWithProviders(<Input ref={inputRef} />);

    const input = screen.getByRole('textbox');

    if (input?.parentElement !== null) {
      fireEvent.click(input?.parentElement);
    }

    if (inputRef.current) {
      inputRef.current.focus();
    }

    expect(input).toHaveFocus();
  });
});
