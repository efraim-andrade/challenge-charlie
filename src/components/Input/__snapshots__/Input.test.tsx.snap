// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - <Input /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 1rem 0.5rem;
  opacity: 0.85;
  background: #E1E1E6;
}

.c0:has(input:focus) {
  opacity: 1;
}

.c1 {
  border: 0;
  width: 100%;
  font-weight: 500;
  background: transparent;
  color: #7A7673;
  font-size: 2rem;
}

.c1:focus {
  outline: 0;
}

.c1::-webkit-input-placeholder {
  color: #8F8F8F;
}

.c1::-moz-placeholder {
  color: #8F8F8F;
}

.c1:-ms-input-placeholder {
  color: #8F8F8F;
}

.c1::placeholder {
  color: #8F8F8F;
}

@media (max-width:600px) {
  .c0 {
    gap: $8;
  }

  .c0 img {
    width: 40;
    height: 40;
  }
}

@media (max-width:600px) {
  .c1 {
    font-size: 1.5rem;
  }
}

<div>
  <div
    class="c0"
  >
    <input
      class="c1"
    />
  </div>
</div>
`;
